src/App.tsx

```diff
@@ -1,24 +1,226 @@
+    })
+      .then((newTodo: Todo) => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTodoTitle('');
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        clearError();
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+
+    setTempTodo({
+      id: 0, userId: USER_ID, title, completed,
+    });
+
+    return promise;
+  };
+
+  const deleteTodo = useCallback((todoId: number) => {
+    setIsLoading(true);
+    setDeletingTodoId(todoId);
+    postService.deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos => filterById(currentTodos, todoId));
+        setDeletingTodoId(null);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        clearError();
+        setDeletingTodoId(null);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [setTodos, clearError]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const updateTodoList = (updatedTodoItem: Todo) => {
+    setTodos(todos.map(todo => {
+      return todo.id === updatedTodoItem.id ? updatedTodoItem : todo;
+    }));
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          todoTitle={todoTitle}
+          setTodoTitle={setTodoTitle}
+          onSubmit={addTodo}
+          isLoading={isLoading}
+          userId={USER_ID}
+          setErrorMessage={setErrorMessage}
+          clearError={clearError}
+          areAllTodosCompleted={areAllTodosCompleted}
+          toggleAll={toggleAll}
+        />
+
+        {visibleTodos && (
+          <TodoList
+            tempTodo={tempTodo}
+            isLoading={isLoading}
+            setIsLoading={setIsLoading}
+            visibleTodos={visibleTodos}
+            deleteTodo={deleteTodo}
+            deletingTodoId={deletingTodoId}
+            toggleTodo={toggleTodo}
+            updateTodoList={updateTodoList}
+            setErrorMessage={setErrorMessage}
+          />
+        )}
+
+        {(todos.length > 0 || tempTodo) && (
+          <Footer
+            todos={todos}
+            setFilter={setFilter}
+            currentFilter={filter}
+            clearCompletedTodos={clearCompletedTodos}
+          />
+        )}
+
+      </div>
+
+      {/* Notification is shown in case of any error */}
+      {/* Add the 'hidden' class to hide the message smoothly */}
```

Comment:
Remove comment. Try to write obvious code with meaningful variable and function naming. Fix all cases